### ISSUE TITLE
Error for points which are out of bounds

### ISSUE THREAD

**Issue by rhattersley** [SciTools/cartopy#517]
Possible point projection error going from Geodetic/lonlat to Geostationary/geos.

Using Cartopy:

``` python
>>> from cartopy import crs
>>> gd=crs.Geodetic()
>>> gd.proj4_init
'+datum=WGS84 +ellps=WGS84 +proj=lonlat +no_defs'
>>> gs = crs.Geostationary()
>>> gs.proj4_init
'+ellps=WGS84 +proj=geos +lon_0=0.0 +lat_0=0 +h=35785831 +x_0=0 +y_0=0 +units=m +no_defs'
>>> gs.transform_point(0, 0, gd)
(0.0, 0.0)
>>> gs.transform_point(10, 0, gd)
(1104210.4459055818, 0.0)
>>> gs.transform_point(40, 0, gd)
(3919923.669560558, 0.0)
>>> gs.transform_point(80, 0, gd)
(5432764.487499327, 0.0)
>>> gs.transform_point(85, 0, gd)
(1.4835298641951802, 0.0)
```

Using cs2cs:

```
cs2cs +datum=WGS84 +ellps=WGS84 +proj=lonlat +no_defs +to +ellps=WGS84 +proj=geos +lon_0=0.0 +lat_0=0 +h=35785831 +x_0=0 +y_0=0 +units=m +no_defs
0 0
0.00    0.00 0.00
10 0
1104210.45  0.00 0.00
40 0
3919923.67  0.00 0.00
80 0
5432764.49  0.00 0.00
85 0
Rel. 4.8.0, 6 March 2012
<cs2cs>: while processing file: <stdin>, line 5
pj_transform(): tolerance condition error
*   * 0.00
```

Note the discrepancy in behaviour when attempting to project a point beyond the visible horizon. Presumably this is linked to the code which [ignores the proj.4 status code](https://github.com/SciTools/cartopy/blob/c022f0401c698487047a480438ee0de11c9d1c4e/lib/cartopy/_crs.pyx#L267).

**Comment by pelson** [comment]
I've updated the title. Essentially, I think you are asking for a way of propagating the proj4 transform errors, right?

**Comment by dopplershift** [comment]
I wonder if this particular case was fixed by my fixes in #1216 ?  Currently this is what I get:
```python
from cartopy import crs
gd = crs.Geodetic()
gs = crs.Geostationary()
gs.transform_point(85, 0, gd)
```
Prints `(nan, nan)`.

**Comment by pelson** [comment]
Let's run with it. I could definitely see us opening up a more general issue about capturing the proj errors, but for now, this issue is as good as resolved. 👍
